### ISSUE TITLE
Does not work with Metrics.NET v3.4 (the latest) prerelease?

### ISSUE THREAD

**Issue by peske** [mnadel/metrics.net.influxdb#8]
Metrics.NET v3.4-pre is not supported? It is true that it is prerelease, but obviously guys that are working on Metrics.NET do not care too much about releases (the last release is 2.16 from March 2015), so we must go with prerelease versions.

Anyway, Metrics.NET.InfluxDB 2.05 beta with Metrics.NET 3.4-pre throws the following exception:

`An unhandled exception of type 'System.MissingMethodException' occurred in Metrics.NET.InfluxDB.dll

Additional information: Method not found: 'Metrics.Reports.MetricsReports Metrics.Reports.MetricsReports.WithReport(Metrics.Reporters.MetricsReport, System.TimeSpan)'.`

Thanks

PS. Honestly, if I were you, I wouldn't know what to do - to stick with the release version of the parent project, or to work on compatibility with prereleases.

**Comment by mnadel** [comment]
In a very unfortunate turn of events, the original author and maintainer of Metrics.NET passed away. His former employer stepped up and has taken ownership of the project within the last month or two. Hopefully they will give it the attention it deserves.

I'm holding out hope that the 3.x series will soon be released as a full version. If not, I'd consider a 3.x branch of this lib that tracks 3.x of Metrics.NET, assuming it has a good ROI. To that end -- have you found 3.4-pre to be stable? What additional features/fixes does it add that you find particularly valuable?

**Comment by mnadel** [comment]
I'm holding out hope that the 3.x series will soon be released as a full version. If not, I'd consider a 3.x branch of this lib that tracks 3.x of Metrics.NET, assuming it has a good ROI. To that end -- have you found 3.4-pre to be stable? What additional features/fixes does it add that you find particularly valuable?

**Comment by mnadel** [comment]
Turns out I needed a 3.x-pre version b/c of the Owin.Metrics package we're using. I've pushed a v3 branch as well as a 3.x package to NuGet.

**Comment by peske** [comment]
Huh, I'm very sad to hear that! RIP...

Regarding stability of 3.x - I can't help you at the moment since I'm just starting to implement Matrix.NET - not enough experience. And obviously since I'm bound to Metrix.NET.InfluxDb - I will not migrate to 3.x before you do.

Regarding 3.x features I need - at the moment nothing. I simply assumed that the newer versions can be improved in terms of stability and performance, so I wanted to go there. Soon I will implement Metrics.NET in MVC/Owin project, and I'll see if some of the new features are needed there. I'll let you know.

Thanks for being so helpful! I wish you luck.

**Comment by peske** [comment]
Hey, 3.0.59  version depends on Metrics.NET v3.4? It means that you've migrated towards their prerelease?

**Comment by mnadel** [comment]
Yep. The day after you mentioned it I found that I needed to do it because we had become dependent on v3 of the Metrics OWIN extension.

**Comment by alhardy** [comment]
Re "implement Metrics.NET in MVC/Owin" I wrote the current Metrics.Net OWIN adapter, OWIN's a thing of the past now (that didn't last long :)), I've written an adapter for ASP.NET Core  if you're interested - https://www.nuget.org/packages/AspNet.Metrics/.

I just hooked the this package up to the MVC sample, works well thanks! You should point to this in the Metrics.NET wiki - https://github.com/Recognos/Metrics.NET/wiki/InfluxDb

Running the latest influx db on the Windows 10 Linux subsystem works like a charm as well :)
